ambry-router/src/main/java/com/github/ambry/router/SimpleOperationTracker.java

```diff
@@ -349,20 +343,45 @@ class SimpleOperationTracker implements OperationTracker {
         remoteOfflineReplicas.forEach(this::addToEndOfPool);
       }
     }
-    totalReplicaCount = replicaPool.size();
-    originatingDcOfflineReplicaCount =
-        getReplicasByState(originatingDcName, EnumSet.of(ReplicaState.OFFLINE)).values().size();
-    originatingDcTotalReplicaCount = allReplicas.stream()
-        .filter(replicaId -> replicaId.getDataNodeId().getDatacenterName().equals(this.originatingDcName))
-        .collect(Collectors.toList())
-        .size();
+    // Count of replicas that will be used for this operation.
+    operationReplicaCount = replicaPool.size();
+
+    // Count of total and offline replicas in all data centers. This will be used later to check if we
+    // should return 503 unavailable instead of 404 not found.
+    if (routerConfig.routerGetEligibleReplicasByStateEnabled) {
+      replicasByStateSnapshot.values().forEach(replicaIds -> totalReplicaCount += replicaIds.size());
+      totalOfflineReplicaCount =
+          replicasByStateSnapshot.getOrDefault(ReplicaState.OFFLINE, Collections.emptyList()).size();
+    } else {
+      totalReplicaCount = partitionId.getReplicaIds().size();
+      // Since we are not looking into replica states, consider offline replica count as 0.
+      totalOfflineReplicaCount = 0;
+    }
+
+    // Count of total and offline replicas in originating data center. This will be used later to check if we
+    // should return 503 unavailable instead of 404 not found.
+    if (routerConfig.routerGetEligibleReplicasByStateEnabled) {
```

Comment:
I wonder why we have 2 implementations and select based on config.  Is there plan to delete the else case?  Looking at AmbryLI, this config value is true for all frontend hosts except one machine "machine-ltx1-app56003.prod.linkedin.com.src" that has a value of false.  Anyone know why this host has special value?  I think we could simplify our code by removing deprecated old code.